### PR TITLE
Doc for Dynamic and Static routing 

### PR DESCRIPTION
**Pull request by weizhouapache**
This is the doc for https://github.com/apache/cloudstack/pull/9470

new page: https://cloudstack-documentation--419.org.readthedocs.build/en/419/adminguide/networking_and_traffic.html#dynamic-and-static-routing

----
📚 Documentation preview 📚: https://cloudstack-documentation--419.org.readthedocs.build/en/419/

### CODE REVIEW

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+An IPv4 subnet for guest networks is created from its parent which is a IPv4 subnet for zone.
+
+There are some global settings which can be set for each account. See below
+
+.. cssclass:: table-striped table-bordered table-hover
+
+================================================= ========================
+Configuration                                       Description
+================================================= ========================
+routed.ipv4.network.cidr.auto.allocation.enabled    whether the auto-allocation of network CIDR for routed network is enabled or not. True by default.
```
Comment (shwstppr) [inline source/adminguide/networking/dynamic_static_routing.rst:88]:
```suggestion
routed.ipv4.network.cidr.auto.allocation.enabled    Whether the auto-allocation of network CIDR for routed network is enabled or not. True by default.
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+- Static. The operators need to add the static routes to the isolated networks or VPCs in the upstream router manually.
+- Dynamic. The AS number will be automatically allocated, and BGP peer sessions will be set up automatically in the VR of the isolated networks or VPCs. The operators need to add the AS number ranges and BGP peers for each zone before creating network with Dynamic routing mode.
+
+
+Manage IPv4 Subnets for Zone
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Like IPv6 prefixes, operators need to configure the IPv4 subnets for zone, which will be eventually used by guest networks.
+
+Supported CloudStack API for operators to manage the IPv4 subnets for zone are:
```
Comment (shwstppr) [inline source/adminguide/networking/dynamic_static_routing.rst:62]:
```suggestion
Supported CloudStack APIs for operators to manage the IPv4 subnets for zone are:
```
Not sure if we need to specify CloudStack each section

**Review by shwstppr — APPROVED** [review]
Changes look good to me but will need functional review during testing

**Comment by shwstppr** [comment]
@blueorangutan docbuild

File: source/adminguide/networking/advanced_zone_config.rst [hunk 3]
```diff
@@ -64,6 +64,8 @@ configure the base guest Network:
    -  **Netmask**: The netmask in use on the subnet the Guest Instances
       will use.
 
+   -  **CIDR Size**: The cidrsize of the subnet the Guest Instances will use. The applies on networks with ROUTED mode only.
```
Comment (weizhouapache) [inline source/adminguide/networking/advanced_zone_config.rst:67]:
```suggestion
   -  **CIDR Size**: The cidrsize of the subnet the Guest Instances will use. Available only when the selected Network offering supports ROUTED mode.
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+Configuration                                       Description
+================================================= ========================
+routed.ipv4.network.cidr.auto.allocation.enabled    Whether the auto-allocation of network CIDR for routed network is enabled or not. True by default.
+routed.ipv4.network.max.cidr.size                   The maximum value of the cidr size for isolated networks in ROUTED mode	
+routed.ipv4.network.min.cidr.size                   The minimum value of the cidr size for isolated networks in ROUTED mode	
+routed.ipv4.vpc.max.cidr.size                       The maximum value of the cidr size for VPC in ROUTED mode	
+routed.ipv4.vpc.min.cidr.size                       The minimum value of the cidr size for VPC in ROUTED mode	
+================================================= ========================
+
+Supported CloudStack API for operators to manage the IPv4 subnets for guest networks are:
```
Comment (weizhouapache) [inline source/adminguide/networking/dynamic_static_routing.rst:95]:
```suggestion
Supported CloudStack APIs for operators to manage the IPv4 subnets for guest networks are:
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+
+Manage IPv4 Routing Firewall
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Users can manage the IPv4 Routing firewalls by navigating to Network -> Guest Networks -> choose a network -> IPv4 Routing Firewall
+
+|routed-ipv4-routing-firewall.png|
+
+Supported CloudStack API for operators to manage the IPv4 Routing firewall rules are:
```
Comment (weizhouapache) [inline source/adminguide/networking/dynamic_static_routing.rst:155]:
```suggestion
Supported CloudStack APIs for operators to manage the IPv4 Routing firewall rules are:
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+
+
+Manage AS number for Dynamic Routing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create network with dynamic routing, operators must add AS number ranges in advance by navigating to Infrastructure -> Zones -> choose a zone -> AS Number.
+
+|dynamic-routing-as-number-ranges.png|
+
+Supported CloudStack API for operators to manage the AS number ranges and AS numbers are:
```
Comment (weizhouapache) [inline source/adminguide/networking/dynamic_static_routing.rst:170]:
```suggestion
Supported CloudStack APIs for operators to manage the AS number ranges and AS numbers are:
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+
+Manage BPG peers for Dynamic Routing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create network with dynamic routing, operators must add BGP peers in advance. Guest networks with Dynamic Routing will connect to all BGP peers the account can access.
+
+|dynamic-routing-bgp-peers.png|
+
+Supported CloudStack API for operators to manage the BGP peers are:
```
Comment (weizhouapache) [inline source/adminguide/networking/dynamic_static_routing.rst:190]:
```suggestion
Supported CloudStack APIs for operators to manage the BGP peers are:
```

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+|dynamic-routing-bgp-peers.png|
+
+Supported CloudStack APIs for operators to manage the BGP peers are:
+
+- **createBgpPeer** : create a BGP peer
+- **dedicateBgpPeer** : dedicate a BGP peer to a domain or an account
+- **deleteBgpPeer** : delete a BGP peer
+- **listBgpPeers** : list BGP peers
+- **releaseBgpPeer** : release a dedicated IPv4 subnet for zone from a domain or an account
```
Comment (weizhouapache) [inline source/adminguide/networking/dynamic_static_routing.rst:196]:
```suggestion
- **releaseBgpPeer** : release a dedicated BGP peer from a domain or an account
```

**Comment by kiranchavala** [comment]
@blueorangutan docbuild

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+
+Dynamic and Static Routing
+-----------------------------
+
+For VMs on Isolated networks, the IP of VMs are not publicly accessible.
+To access the VMs from the Internet, Users need to create Load balanceing rules,
```
Comment (kiranchavala) [inline source/adminguide/networking/dynamic_static_routing.rst:21]: Load balancing  spelling

File: source/adminguide/networking/dynamic_static_routing.rst [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+
+|dynamic-routing-change-network-bgp-peers.png|
+
+
+Create VPC with Dynamic Routing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The creation of VPC with Dynamic routing is almost as VPC with static routing. CloudStack will allocate an AS number to the VPC, and 
+- If the VPC owner does not have dedicated BGP peers, or account setting "use.system.bgp.peers" is set to true, configure BGP sessions in the VPC VR to connect to all BGP peers the VPC owner can access.
+- If the VPC owner has dedicated BGP peers, and account setting "use.system.bgp.peers" is set to false, configure BGP sessions in the VOC VR to connect to all dedicated BGP peers of the domain and the VPC owner.
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply source/adminguide/networking/dynamic_static_routing.rst]: thanks, fixed

**Review by kiranchavala — APPROVED** [review]
LGTM

**Pull request merged**